### PR TITLE
Really move verify tests to Python 3

### PR DESCRIPTION
**Pull request by martinpitt**
- [x] fix artifact building and sit on failures (`checkSuccess` is always true even on failures)
 - [x] fix check-connection regressions
 - [x] fix other test regressions
 - [x] fix `check-ostree OstreeCase.testRemoteManagement`
 - [x] introduce `_PY3` in testlib.py (PR #9582)

### CODE REVIEW

**Comment by martinpitt** [comment]
@mvollmer: For debugging your tests, either cherry-pick "test: Fix checkSuccess() for Python 3" here, or simply run the test through python2 for the time being (`python2 test/verify/check-foo ...`).

**Comment by mvollmer** [comment]
Great work!

**Comment by martinpitt** [comment]
Only one well-known flake, I think this is green enough. (We can't run tests ATM)

File: test/verify/check-connection [hunk 1]
```diff
@@ -255,13 +255,13 @@ class TestConnection(MachineCase):
 
         # login handler: wrong password
         headers = m.execute("curl -s --head --header 'Authorization: Basic {}' http://172.27.0.15:9090/cockpit/login".format(
-            base64.b64encode(b"admin:hahawrong")))
+            base64.b64encode(b"admin:hahawrong").decode()))
```
Comment (allisonkarlitskaya) [inline test/verify/check-connection]: b64encode returns bytes?!  this is some kind of awful :(

(I checked, and indeed, it seems that there is no better way to handle this)

File: test/verify/check-login [hunk 2]
```diff
@@ -286,9 +286,10 @@ account    required     pam_succeed_if.so user ingroup %s""" % m.get_admin_group
         self.allow_restart_journal_messages()
 
     def curl_auth(self, url, userpass):
-        header = "Authorization: Basic " + base64.b64encode(userpass)
+        header = "Authorization: Basic " + base64.b64encode(userpass.encode()).decode()
```
Comment (allisonkarlitskaya) [inline test/verify/check-login]: lol

File: bots/tests-policy [hunk 3]
```diff
@@ -221,6 +221,9 @@ def normalize_traceback(trace):
     # replace noise in SELinux violations
     trace = re.sub('audit\([0-9.:]+\)', 'audit(0)', trace)
     trace = re.sub(r'\b(pid|ino)=[0-9]+ ', r'\1=0 ', trace)
+
+    # in Python 3, testlib.Error is shown with namespace
+    trace = re.sub('testlib\.Error', 'Error', trace)
```
Comment (allisonkarlitskaya) [inline bots/tests-policy]: I'd strongly consider normalising this in the other direction (and updating the naughties).  It seems that py3 is the expected norm now, so why not treat it that way?

Comment (martinpitt) [reply bots/tests-policy]: I filed issue #9591 about it. I'm still hoping that the actually printed format can be fixed to just `Error:` again, I just didn't see a good way of doing this. I don't want to change all the naughties until this settled down and we also port the selenium, avocado, and container tests.

**Pull request merged**